Tests/MessagingInApp/Gist/Managers/MessageQueueManagerTest.swift

```diff
@@ -64,6 +64,45 @@ class MessageQueueManagerTest: UnitTest {
         XCTAssertEqual(actualMessages, [givenMessage2, givenMessage1, givenMessage3])
     }
 
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMatch_expectInlineMessage() {
+        let givenElementId1 = String.random
+        let givenElementId2 = String.random
+        let givenElementId3 = String.random
+        let currentRoute = String.random
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId1, priority: 1)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId2, priority: 0)
+        let givenMessage3 = Message(elementId: givenElementId3, priority: 0)
+
+        manager.localMessageStore = [
+            "1": givenMessage1,
+            "2": givenMessage2,
+            "3": givenMessage3
+        ]
+        Gist.shared.setCurrentRoute(currentRoute)
+        let actualMessages = manager.getInlineMessages(forElementId: givenElementId1)
+        XCTAssertEqual(actualMessages, [givenMessage1])
+    }
+
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMisMatch_expectEmptyArray() {
+        let givenElementId1 = String.random
+        let givenElementId2 = String.random
+        let givenElementId3 = String.random
+        let currentRoute = String.random
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId1, priority: 1)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId2, priority: 0)
+        let givenMessage3 = Message(elementId: givenElementId3, priority: 0)
+
+        manager.localMessageStore = [
+            "1": givenMessage1,
+            "2": givenMessage2,
+            "3": givenMessage3
+        ]
+        let actualMessages = manager.getInlineMessages(forElementId: givenElementId1)
+        XCTAssertEqual(actualMessages, [])
+    }
+
```

Comment:
Great tests so far. I think there is 1 test missing: given the SDK has a route set (`Gist.shared.setCurrentRoute(currentRoute)`) but the route not does match any of the page rules set on messages.

Reply:
Another use case to test: given the SDK has a route set (Gist.shared.setCurrentRoute(currentRoute)) but a `Message` does not have a page rule added to it (`Message(pageRule: nil)`), you expect that Message to be returned.

Reply:
Addressed. ✅